crates/compiler/mono/src/code_gen_help/equality.rs

```diff
@@ -660,8 +660,7 @@ fn eq_list<'a>(
     let arena = root.arena;
 
     // A "Box" layout (heap pointer to a single list element)
-    let box_union_layout = UnionLayout::NonNullableUnwrapped(root.arena.alloc([elem_layout]));
-    let box_layout = layout_interner.insert(Layout::Union(box_union_layout));
+    let box_layout = layout_interner.insert(Layout::Boxed(elem_layout));
```

Comment:
apparently I forgot to update this to use box some time ago

Reply:
oh yeah I remember we talked about that